lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java

```diff
@@ -81,8 +86,11 @@ private static LogicalType fromSchemaImpl(Schema schema, boolean throwErrors) {
         logicalType.validate(schema);
       }
     } catch (RuntimeException e) {
+      LOGGER.debug("Invalid logical type found", e);
       if (throwErrors) {
         throw e;
+      } else {
+        LOGGER.warn("Ignoring invalid logical type for name: {}", typeName);
```

Comment:
Minor clarification: By "all the lines" I meant the single `logicalType = null;` line. :)